cterasdk/core/files/collaboration.py

```diff
@@ -0,0 +1,255 @@
+            _update_share_param(share_param, accounts_to_keep)
+            logging.getLogger().info(
+                'Removing share recipients. %s',
+                {'path': str(path.relativepath), 'recipients': [str(account) for account in accounts_to_remove]}
+            )
+            ctera_host.execute('', 'shareResource', share_param)
+    else:
+        logging.getLogger().info('Share has no recipients. %s', {'path': str(path.relativepath)})
+    return accounts_to_remove
+
+
+def get_share_info(ctera_host, path):
+    return ctera_host.execute('', 'listShares', path.encoded_fullpath())
+
+
+def share(ctera_host, path, recipients, as_project, allow_reshare, allow_sync):
+    if len(path.relativepath.parts) > 1:  # The shared path is not a cloud folder. Therefore, the following attrs aren't customizable.
+        as_project = False
+        allow_sync = False
+
+    valid_recipients = _obtain_valid_recipients(ctera_host, path, recipients)
+    if valid_recipients:
+        share_param = _create_share_param(path.fullpath(), as_project, allow_reshare, allow_sync)
+        for recipient in valid_recipients:
+            _add_share_recipient(share_param, recipient)
+
+        logging.getLogger().info(
+            'Sharing item. %s',
+            {'path': str(path.relativepath), 'recipients': [str(recipient) for recipient in valid_recipients]}
+        )
+        ctera_host.execute('', 'shareResource', share_param)
+        return valid_recipients
+    logging.getLogger().warning('Resource was not shared. Could not find valid recipients. %s', {'path': str(path.relativepath)})
+    return valid_recipients
+
+
+def add_share_recipients(ctera_host, path, recipients):
+    share_info = get_share_info(ctera_host, path)
+    current_accounts = _obtain_current_accounts(share_info)
+    valid_recipients = _obtain_valid_recipients(ctera_host, path, recipients)
+    share_param = _create_share_param(path.fullpath(), share_info.teamProject, share_info.allowReshare, share_info.shouldSync)
+    _update_share_param(share_param, share_info.shares)
+    accounts_added = []
+    if valid_recipients:
+        for recipient in valid_recipients:
+            if recipient.account not in current_accounts:
+                logging.getLogger().debug('Found new share recipient. %s', {'recipient': str(recipient.account)})
+                _add_share_recipient(share_param, recipient)
+                accounts_added.append(recipient)
+            else:
+                logging.getLogger().debug('Share recipient already exists. %s', {'recipient': str(recipient.account)})
+        if accounts_added:
+            logging.getLogger().info(
+                'Adding share recipients. %s',
+                {'path': str(path.relativepath), 'recipients': [str(recipient.account) for recipient in accounts_added]}
+            )
+            ctera_host.execute('', 'shareResource', share_param)
+            return accounts_added
```

Comment:
As in the above method. Put the print in an else clause and keep in single return (at least within the "if valid_recipients" clause

Reply:
Done